utils/dateUtils.ts

```diff
@@ -0,0 +1,36 @@
+export const formatWithTimeAndDate = (isoString: Date) => {
+    if (isoString) {
+        // Parse the ISO date string into a Date object
+        const date = new Date(isoString);
+
+        // Define arrays for month names
+        const months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
+
+        // Extract date components
+        const hours = date.getUTCHours();
+        const minutes = date.getUTCMinutes();
+        const day = date.getUTCDate();
+        const month = date.getUTCMonth(); // Month is 0-based
+        const year = date.getUTCFullYear();
+
+        // Format hours and minutes
+        const formattedHours = hours.toString().padStart(2, '0'); // Add leading zero if needed
+        const formattedMinutes = minutes.toString().padStart(2, '0'); // Add leading zero if needed
+
+        // Format the final string
+        return `${formattedHours}:${formattedMinutes} on ${months[month]} ${day}, ${year}`;
+    } else {
+        return 'No date';
+    }
+}
+
+export const formatShortMonthDayYear = (isoString: Date) => {
+    if (isoString) {
+        const date = new Date(isoString);
+        const options: Intl.DateTimeFormatOptions = { year: 'numeric', month: 'short', day: 'numeric' };
+        return date.toLocaleDateString('en-US', options);
+    } else {
+        return 'No date';
+    }
+
+}
```

Comment:
Is there a reason we aren't using something like "momentjs"[1] for this? It makes working with date and time much easier and consistent.

[1] https://momentjs.com/docs/

Reply:
yes agree with this.

Reply:
@andrewebdev and @briri, do we need to internationalize dates in the app? If not, I think, for now, we can leave this dateUtils.ts in place, and discuss other options as we progress in the project.

I read up on Moment, and the docs for Moment are discouraging developers from using it because it is outdated and because of its size. (https://momentjs.com/docs/).  The site also mentions that there is a proposal to include "[Temporal](https://tc39.es/proposal-temporal/docs/index.html)" in the JavaScript library to handle dates and times.

Reply:
Oh that's news to me! Thank you @jupiter007 ! I'll go read up on this and look at our options. I'm still of the opinion that we should avoid working with datetime ourselves and rather let a good library take care of it. I refer to that YT video by Tom Scott [1] :)

[1] https://www.youtube.com/watch?v=-5wpm-gesOY

Reply:
I am planning to store everything in UTC time in the database. I think that user's will want to see local time in the UI.